### DISCUSSION TITLE
Is Deno relatively unsuccessful, failing to gain traction?

### DISCUSSION THREAD

**Discussion by lukepuplett** [denoland/deno#16991] [General]
Hello all,

I come with an awkward question. I have been coding for 35 years and most recently I work on ASP.NET Core and I make do with KnockoutJS for more interactive UI.

A few weeks ago I thought I'd learn a totally different way of working. I follow enough in frontend development to know that Deno is quite new and since I'm used to large, statically typed codebases, it's TS native support seemed cool.

I've made a Hello World and even implemented the Google OAuth dance by hand. Life's good.

However, I've then started to look at the newer breed of JS frameworks like Svelte and Remix, however, I have been completely confounded by how they work, from the point of view of an HTTP request arriving and being responded to.

Remix is a little easier to understand, but it talks about Node. And in fact, all the tooling and really everything talks only about Node.

The thing is, because I'm so new to all this, I don't know what "the word on the street is" and so it seems really odd that a set of newest and coolest frameworks ignore Deno.

I'm completely stuck to be honest and have genuinely had migraines and sleepless nights with the intensity of the learning curve, esp. trying to dig deeper and deeper into a world that's alien to me of bundlers and tools and plugins and trying to look behind the curtain.

I'm reaching the conclusion that no one uses Deno or plans to. Or perhaps I should ignore the frameworks for now, update KnockoutJS to AlpineJS and wait for Jake Archibald's work on page transitions to bear fruit and potentially cause a resurgence in MPAs.

I appreciate that this is a tough question, but it's an important one. What are your thoughts?

**Comment by nhrones** [comment]
You should bring this conversation to the discord channel.
https://discord.com/channels/684898665143206084/684898665151594506

**Comment by MrPossumz** [comment]
I think your reference point you're using to judge the success of Deno is a bit faulty. Svelte, for example, was initially released in 2016 which was a good two years before Deno.

Node.js has traditionally been the main runtime for JS and TS tooling which is why it's nearly universal. It has a much bigger user base and market share so a small team working on a new idea is going to want to maximize the return on their efforts by targeting them over other options like Deno or Bun. 

The Deno team has made impressive strides to provide compatibility with the npm and node environment. They also, in my opinion, provide a much nicer DX so I believe we'll see adoption continue to grow. There is a LOT to learn here but funny enough, Deno helps to take care of a decent amount of that overhead (like trying to setup a typescript project in Node.js!) and lets you really jump into coding.

I do agree that it still has some issues with providing simpler plug and play solutions or configurations for generating web UI using the more popular frameworks, but those frameworks are __complicated__ and there's a lot involved with making it work.

**Comment by lukepuplett** [comment]
For me, with no prior JS stack experience, Deno was a no brainer but I think I must have turned up just as those "strides" had been taken. I read the docs fully before starting out and it seemed that interop with the Node world was probably going to be fine.

So I've been surprised that the frameworks, especially the newer CSR+SSR ones that I'd imagine people would be looking to use on greenfield projects, have not got Deno and Node as equal weighted choices from the start. My guess is that maybe the knowledge of the npm compatibility work hasn't permeated yet and the people working on these frameworks need a little nudge to get them to look again at first class Deno support from the get-go.

I eventually started down the Fresh route, though I'm not sure if that's a "serious" framework or a demonstration. I watched a session at a Remix event with Ryan Dahl announcing it, and he seemed to act like it wasn't a serious thing. He may just have been playing it down given that he'd apparently come to a Remix event to shill his own framework but I don't know. It's all too new to me.

**Comment by itsdouges** [comment]
I've found Deno awesome, starting to use it after the npm stable announcement. TS working without any funny business, the VSCode extension works a treat (including showing available tests to run, great!), even the module preference over packages has been a breathe of fresh air. I think what's missing is developers just building stuff with Deno, better docs, and fixing bugs. NPM compat is huge, now there's the final stretch of getting it to work seamlessly with Deno (e.g npm based bundlers!).

**Comment by guest271314** [comment]
I prefer `deno` executable to `node` executable. I use `qjs` (QuickJS) executable the most locally for JavaScript engine at ~1MB executable file size after `strip` and RSS at ~3MB during usage.

**Comment by o-t-w** [comment]
You can use Remix with Deno, although troubleshooting and finding documentation will not be as good as for Node as it's still less popular.
